apps/site/src/pages/blog/[slug].astro

```diff
@@ -1,13 +1,41 @@
 ---
-import '../../styles/global.css';
+import type { GetStaticPaths } from "astro";
+import { getCollection } from "astro:content";
+import Layout from "~/layouts/Layout.astro";
 
-import { getEntry } from 'astro:content';
+export const getStaticPaths = (async () => {
+  const posts = await getCollection("blog");
+  return posts.map((post) => ({
+    params: {
+      slug: post.slug,
+    },
+    props: {
+      post,
+    },
+  }));
+}) satisfies GetStaticPaths;
```

Comment:
What is the reason to opt for prerendering instead of on-demand rendering?

Reply:
Same as my comment above about server => hybrid